### ISSUE TITLE
Error when deploying to AWS S3

### ISSUE THREAD

**Issue by derwaldgeist** [yeoman/generator-webapp#441]
I've set-up the S3 deployment as described in the docs. I also entered my key, secret and bucket name into the gulp task. Yet, I am getting this error message:

```
/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/request.js:30
            throw err;
            ^

MissingRequiredParameter: Missing required key 'Bucket' in params
    at ParamValidator.fail (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/param_validator.js:50:37)
    at ParamValidator.validateStructure (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/param_validator.js:61:14)
    at ParamValidator.validateMember (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/param_validator.js:88:21)
    at ParamValidator.validate (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/param_validator.js:34:10)
    at Request.VALIDATE_PARAMETERS (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/event_listeners.js:89:42)
    at Request.callListeners (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/sequential_executor.js:105:20)
    at callNextListener (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/sequential_executor.js:95:12)
    at /Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/event_listeners.js:75:9
    at finish (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/config.js:267:7)
    at /Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/config.js:307:9
    at resolveNext (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/credentials/credential_provider_chain.js:84:9)
    at /Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/credentials/credential_provider_chain.js:97:11
    at /Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/credentials.js:123:23
    at /Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/credentials/ec2_metadata_credentials.js:66:7
    at callbacks (/Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/metadata_service.js:103:9)
    at /Users/waldgeist/Documents/Development/htdocs/aiddogs/website/node_modules/aws-sdk/lib/metadata_service.js:108:16
```

Could it be that I have to provide the bucket name in a certain syntax? I am also wondering why the config params do not contain a region (I'm using 'eu-west-1').

**Comment by derwaldgeist** [comment]
After digging into the documentation of ```gulp-awspublish``` and the AWS SDK, I found out that it works with this configuration:
```
  const publisher = $.awspublish.create({
    "accessKeyId": "...",
    "secretAccessKey": "...",
    "region": "...",
    params: {
      Bucket: '...'
    }
  });
```
which is quite different from what the current recipe states. Maybe you would like to update it. I am also asking myself if there's a way to use the AWS recommendation to keep the credentials separate. At least it would be nice if they were located in a separate JSON file, so I do not have to include it in my versioning system, which is generally a bad practice.

The possibilities are described here:
http://docs.aws.amazon.com/AWSJavaScriptSDK/guide/node-configuring.html
but I am not sure how I can access AWS.config.loadFromPath.

**Comment by derwaldgeist** [comment]
Got the file import working using this code:
```
const fs = require("fs");

gulp.task('deploy', ['build'], () => {
  // create a new publisher
  const credentials = JSON.parse(fs.readFileSync('aws-credentials.json', 'utf8'));
  const publisher = $.awspublish.create(credentials);

  // define custom headers
  const headers = {
    'Cache-Control': 'max-age=315360000, no-transform, public'
  };

  return gulp.src('dist/**/*.*')
    .pipe(publisher.publish(headers))
    .pipe(publisher.sync())
    .pipe(publisher.cache())
    .pipe($.awspublish.reporter());
});
```
BTW: The awspublish docs also recommend the following pipe:
```
.pipe(awspublish.gzip({ ext: '.gz' }))
```
Maybe this compression speeds things up? And they are not using the sync().

**Comment by silvenon** [comment]
I believe this has just been fixed by @jkatsnelson in #622. Let me know if this is still an issue and I'll gladly reopen.
